### ISSUE TITLE
search can't search multiple files

### ISSUE THREAD

**Issue by bean5** [overleaf/overleaf#139]
When I used a template file that has a lot of files, it is difficult to find and replace things like Author, or Title, unless I can search for the word.

**Comment by Dartspacephysiker** [comment]
Is this still not possible?

**Comment by ChristianEllis** [comment]
+1

**Comment by andrewjong** [comment]
Overleaf web was archived. Please reopen this issue.

**Comment by ezhang7423** [comment]
Thumbs up, would greatly appreciate this.

**Comment by amirootyet** [comment]
We would greatly appreciate this features. (Writing this as I'm about to miss a submission deadline because I'm looking for a phrase that the reviewers wanted me to address, but I can't find it through the 25+ pages of my manuscript on Overleaf.)

**Comment by mcever** [comment]
You can always download the entire project and search on your local device. Really unfortunate that it must be done this way, but it's a temporary workaround for anyone visiting this page in the meantime until Overleaf implements this critical function.

**Comment by JoeyPriceless** [comment]
+1

It's frankly embarrassing that this feature is missing for a paid text editor.

**Comment by ucfjimg** [comment]
It wouldn't be so bad but searching in the compiled PDF doesn't work reliably either, unless you first download it and reopen it in another tab.

**Comment by eahenle** [comment]
@ucfjimg searching the PDF doesn't work _at all_ if you're looking for non-rendered characters (comments, commands)

When is this feature coming? @JoeyPriceless is absolutely correct--this should have been implemented almost a decade ago, when the issue was first opened!

**Comment by lrperlmu** [comment]
Any estimate on when this feature will be implemented?

**Comment by cmprmsd** [comment]
It's on my agenda 2024 when this feature arrives in the self hosted community version. 😁

**Comment by MiladInk** [comment]
it would be really nice if implemented.

**Comment by wltjr1007** [comment]
+1

**Comment by bramgrooten** [comment]
+1, would be very useful to be able to replace a string across all files in a project (without having to go through all files one by one)

**Comment by wiegerw** [comment]
Search and replace across multiple files is an essential feature. It is very cumbersome that you have to manually open all files in order to rename an identifier.

**Comment by SandroAlex** [comment]
Search and replace across multiple files is very useful for students and researchers who use overleaf.

**Comment by Gkreindler** [comment]
+1 
This should be a feature very soon. I just got stuck and wasted a lot of time because this doesn't exist.

**Comment by sebffischer** [comment]
+1

**Comment by WJPeace1** [comment]
+1

**Comment by jackal1-66** [comment]
Indeed, this could be extremely time saving and very valuable as a feature

**Comment by cmprmsd** [comment]
If only they would listen to community feedback... 
There is a lot of stuff ongoing in development but I think they should do the essentials first before driving deep into the visual editor improvements. 😪

**Comment by julou** [comment]
+1

**Comment by vmichals** [comment]
+1

**Comment by borgr** [comment]
Happy new year, it has been 10 year since this was opened. 
By the above many thousands of us are eager to get this feature

**Comment by sourav22899** [comment]
+1. This has been a highly requested essential feature. People who use overleaf a lot, hardly ever work on a single big tex file. Manually doing find and replace across a bunch of tex files is simply a time waste.

**Comment by Musicminion** [comment]
+1. This has been a highly requested essential feature. Similar products like [**Texpage**](https://texpage.com/) **has finished this Feature** .

**Comment by nik-ko** [comment]
+1. This is a feature I so often missed in overleaf.
A workaround that I use is the to synchronize the overleaf project with github. There you can comfortably search in all files within the whole repository.

**Comment by wspear** [comment]
I was collaborating over a larger, structured project in Overleaf for the first time recently and was unpleasantly surprised to find that whole-project searching wasn't possible. This is a major omission.

**Comment by maayanorner** [comment]
Not to be rude, but that's pretty weird this feature does not exist.

**Comment by 1henrypage** [comment]
Can you please add this feature? Every editor has this in some capacity.

**Comment by Iaotle** [comment]
Agreed, this needs to be a thing.

**Comment by ryanwwest** [comment]
Research collaborators wonder why I put everything in a single .tex file. I know this method has its disadvantages, but I prefer this to dealing with opening and individually searching every .tex file whenever I want to find and replace a string. But this makes Overleaf a lot less usable as project size increases.

**Comment by aminst** [comment]
+ 1, this could be a very useful feature.

**Comment by SyedShaQutub** [comment]
I recently purchased a year-long subscription to Overleaf, only to discover that the Overleaf team has been struggling to resolve a fundamental issue for over a decade. Sad story
@jdleesmiller

**Comment by Musicminion** [comment]
@jdleesmiller Sorry to interrupt, But I want to mention, this is a feature crossing 10 years. What makes it so difficult to imply a feature searching in a project?

The following is a screen shot of texpage, it is cheaper then overleaf, but it can search word in a project. 
![image](https://github.com/overleaf/overleaf/assets/84625273/c6f7e4e8-8c49-4f6c-b6bd-7576960cc47e)

I checked the implement of texpage, they use socket API to search project. when user search words, it respones with all the hitted documents.

Hopefully this can be helpful to your development.

**Comment by nik-ko** [comment]
Thanks, I've just cancelled my overleaf subscription.

**Comment by wltjr1007** [comment]
> @jdleesmiller Sorry to interrupt, But I want to mention, this is a feature crossing 10 years. What makes it so difficult to imply a feature searching in a project?
> 
> The following is a screen shot of texpage, it is cheaper then overleaf, but it can search word in a project. ![image](https://private-user-images.githubusercontent.com/84625273/323200222-c6f7e4e8-8c49-4f6c-b6bd-7576960cc47e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzNTc2MjYsIm5iZiI6MTcxMzM1NzMyNiwicGF0aCI6Ii84NDYyNTI3My8zMjMyMDAyMjItYzZmN2U0ZTgtOGM0OS00ZjZjLWI2YmQtNzU3Njk2MGNjNDdlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE3VDEyMzUyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwOTA4NjE5NzE2ODliNWQ1YWMyMGJmZjExZDFkNjE4NzcyZjkxZjFmOTEyMDEzNWRmMWFjZGVlYmYxNTNhNTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.csoGbXRKX4lB-kdciUKchIV-G2yq5hK0_N8no3ZMowY)
> 
> I checked the implement of texpage, they use socket API to search project. when user search words, it respones with all the hitted documents.
> 
> Hopefully this can be helpful to your development.

Nice! giving texpage a try :)

**Comment by ramon-san** [comment]
Agreed, this feature is important. @das7pad anything that could be done about this? Right now had to download all files to VS Code to do this. If this is not possible it would also be good to link directly to Overleaf project through SSH or FTP to allow use of VS Code for text editing—if this is already a possibility would appreciate the docs to do so.

**Comment by ayaka-lab** [comment]
I have gathered some of this issue's comments, and send them all to overleaf support, now they give me a response.

They already know the importance of this feature, and they are aware of this in the future. I also asked the specific time of the release of this feature(search in projects), but they are not sure yet.

I checked the commit of overleaf in recent 3 years, it seems that they are busy with admin panel, sso, and enterprise related chores, and **specially the Visual Editor that I never used**. They changed the past ace-editor to codemirror-6 editor, even if some people are never aware of this. I don't need to know how many studends are using overleaf in a school, how many projects they have covered, even if I am a manager of a school. All in all, those feature is **NOT** what we need.

I admit that some feature in visual editor may be helpful for beginner indeed, but I just want to search some word in a project. As an software-engineer, you may need to re-consider the priority of your overleaf's software requirements.

We just want an easy-to-use editor for latex, isn't it? Maybe your development teams have gone so far today, that you have forgotten why you started your Overleaf project(or sharelatex in the past time) at the very first time.

**Comment by Musicminion** [comment]
Hey gays, recently I found a workaround for search in an overleaf project! We can use [vscode overleaf plugin](https://github.com/iamhyc/Overleaf-Workshop), it can support overleaf-ce, server-pro or even overleaf-common in [www.overleaf.com](www.overleaf.com)


![image](https://github.com/overleaf/overleaf/assets/84625273/784ca1a2-55df-48f6-be49-2402b6b2d05d)

And you can even use copilot, free for students with GitHub Student Pack, just go tab!
![image](https://github.com/overleaf/overleaf/assets/84625273/0567b699-6e73-44a7-98c2-b49dc79476f3)

Of course, chat with copilot is also possible.
![image](https://github.com/overleaf/overleaf/assets/84625273/d1144bf5-06bf-49c8-a19a-446432d68acd)

**Comment by borgr** [comment]
I submitted it to https://www.overleaf.com/contact, it seems that is the current right way to do so (probably 10 years ago it wasn't). Feel free to do the same and write here when you do.

**Comment by isCopyman** [comment]
> Hey gays, recently I found a workaround for search in an overleaf project! We can use [vscode overleaf plugin](https://github.com/iamhyc/Overleaf-Workshop), it can support overleaf-ce, server-pro or even overleaf-common in [www.overleaf.com](www.overleaf.com)
> 
> ![image](https://private-user-images.githubusercontent.com/84625273/327940715-784ca1a2-55df-48f6-be49-2402b6b2d05d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0Mjc2MzUsIm5iZiI6MTcxNzQyNzMzNSwicGF0aCI6Ii84NDYyNTI3My8zMjc5NDA3MTUtNzg0Y2ExYTItNTVkZi00OGY2LWJlNDktMjQwMmI2YjJkMDVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAzVDE1MDg1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4NzhiM2NhYzRjODY1YTlhZjNkNjhjODdlOTRiMmQ3MmVmZTFkN2MzNjQyZjBlZjhjYmI4ZGI0NTFmMjA5ODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IWWC0CSE4TisrfmIuXgVooBJB4lwYhJYFPPIeIkHvHI)
> 
> And you can even use copilot, free for students with GitHub Student Pack, just go tab! ![image](https://private-user-images.githubusercontent.com/84625273/327940797-0567b699-6e73-44a7-98c2-b49dc79476f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0Mjc2MzUsIm5iZiI6MTcxNzQyNzMzNSwicGF0aCI6Ii84NDYyNTI3My8zMjc5NDA3OTctMDU2N2I2OTktNmU3My00NGE3LTk4YzItYjQ5ZGM3OTQ3NmYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAzVDE1MDg1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5YWRjMDRmMGVmOGY5YTA2NWMyZGYzMTQzOTliOWM4YzA5MDY2YzUwOTUzNTMzZjBjZTFjMGUxNjg0MTY1ZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JRH840LJHzu7bR2oiHFzbdxsTqERiDXAiDrdwstttSo)
> 
> Of course, chat with copilot is also possible. ![image](https://private-user-images.githubusercontent.com/84625273/327940926-d1144bf5-06bf-49c8-a19a-446432d68acd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0Mjc2MzUsIm5iZiI6MTcxNzQyNzMzNSwicGF0aCI6Ii84NDYyNTI3My8zMjc5NDA5MjYtZDExNDRiZjUtMDZiZi00OWM4LWExOWEtNDQ2NDMyZDY4YWNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAzVDE1MDg1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwZjMwNWI1N2IwY2U4OTAwZGQyNzA2YWQyOTBlYjVhYTQ0MmNjYzE0NDU0NjMxZTdkOTk4MTkxMjhkMDU3ZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7kq14YDOMbQ1WSoAXFGzf9omI7s-zaPsHcVMTSvzpCs)

This is really great, it perfectly solves the problem

**Comment by SyedShaQutub** [comment]
This is indeed great. It works like a charm. I hope this should also solve some latency issues related to the overload of tabs in Chrome.

**Comment by SyedShaQutub** [comment]
@Musicminion @isCopyman I tested the VSCode plugin, and while it's good, it still encounters the same issue. It won't search in all the files until you've opened all the files in the editor each time you attempt to log in. Are you experiencing the same issue?

**Comment by isCopyman** [comment]
yes, I can only search the opened files. Nevertheless, it still meets the majority of needs.



---Original---
From: ***@***.***&gt;
Date: Thu, Jun 6, 2024 20:13 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [overleaf/overleaf] search can't search multiple files (#139)




 
@Musicminion @isCopyman I tested the VSCode plugin, and while it's good, it still encounters the same issue. It won't search in all the files until you've opened all the files in the editor each time you attempt to log in. Are you experiencing the same issue?

**Comment by rchtgpt** [comment]
this feature will be a godsend

**Comment by fewagner** [comment]
+1, I would also need this feature 
@henryoswald @das7pad @briangough 
if this issue was a child, it would move on to middle school this year

**Comment by xiyuzhai** [comment]
This is a joke. All things around latex are nothing but professional convenient. How does it come to this.

**Comment by xiyuzhai** [comment]
I guess I will tell the story to my grandson.

**Comment by cmprmsd** [comment]
@mserranom @jdleesmiller 
This feature would be a great xmas present or something to add to the roadmap for 2025 as it's in high demand by a variety of users. 
If you don't plan to add this feature, closing as not planned would be a closing statement as well.

**Comment by jdleesmiller** [comment]
We are planning to address this. I'm afraid it will not be under the tree this year, but it is on the roadmap for 2025.

**Comment by borgr** [comment]
Sometimes mountains go skiing!

**Comment by jacky-liang** [comment]
how is this still not supported shouldn't be that hard...

**Comment by aeaton-overleaf** [comment]
We're pleased to announce that it's now possible to search across all the editable files in an Overleaf project. This feature is currently available to users who are part of the [Beta Program](https://www.overleaf.com/beta/participate).

<div>
<img width="289" alt="Image" src="https://github.com/user-attachments/assets/7e566266-b2d5-4711-bf41-b3c50eeaeaf4" />
</div>

<div>
<img width="289" alt="Image" src="https://github.com/user-attachments/assets/5cd8f30e-50f5-4b99-9146-2d681702aa61" />
</div>

**Comment by aeaton-overleaf** [comment]
> is this possible for overleaf-ce version?

This should eventually be available in the Community Edition, yes.

**Comment by hvstms** [comment]
Fantastic! Thank you!🙏🏽

**Comment by jiangsn** [comment]
> We're pleased to announce that it's now possible to search across all the editable files in an Overleaf project. This feature is currently available to users who are part of the [Beta Program](https://www.overleaf.com/beta/participate).
> 
> <img alt="Image" width="289" src="https://private-user-images.githubusercontent.com/75253002/427583617-7e566266-b2d5-4711-bf41-b3c50eeaeaf4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxMTU5NzIsIm5iZiI6MTc0MzExNTY3MiwicGF0aCI6Ii83NTI1MzAwMi80Mjc1ODM2MTctN2U1NjYyNjYtYjJkNS00NzExLWJmNDEtYjNjNTBlZWFlYWY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI3VDIyNDc1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlZWEwYjdhNzBjMWYxNGUxZGRhNDUxNTU0Y2Y1NjI2NTg4OTg4MDM5MTlhZGQyMDkzMjI0M2Y5MjBmZTA3ZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.upCL_dz0nWh4Yx39TfJraJ9k8nHQ4aFywpZ2te8qjfY">
> <img alt="Image" width="289" src="https://private-user-images.githubusercontent.com/75253002/427585615-5cd8f30e-50f5-4b99-9146-2d681702aa61.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxMTU5NzIsIm5iZiI6MTc0MzExNTY3MiwicGF0aCI6Ii83NTI1MzAwMi80Mjc1ODU2MTUtNWNkOGYzMGUtNTBmNS00Yjk5LTkxNDYtMmQ2ODE3MDJhYTYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI3VDIyNDc1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhMDUwNjU4YjQ5NDE5OTdlNDU0NzBkNTc1Yjg2NTlhYWYyZTk4MWFmZTI0ODMzNjI4YTFmMTU2YTkxZTYxOTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vRQqBw-7pEtFzEHlUSv62Um8yok0whZJTdGZ1XWBXaQ">

Finally! Now I don't have to download the project and search locally. That's awesome!

**Comment by Mark-htmlgogogo** [comment]
I'd like to perform a search-and-replace operation across all files. Is that feasible?

**Comment by mserranom** [comment]
> I'd like to perform a search-and-replace operation across all files. Is that feasible?

I'm afraid that's not possible. If you're using Linux/Mac and have Dropbox/Github/Git Integration enabled in your overleaf account, you can checkout out the project and the following command, which will replace all the entries of the word `old` with the word `new`:

```
find . -type f -name "*.tex" -exec sed -i 's/old/new/g' {} +
```

**Comment by carlosgmartin** [comment]
@mengfan-ma I've opened a request for that feature here: https://github.com/overleaf/overleaf/issues/1400.
